src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -79,5 +80,21 @@ public static async Task ReadPrimitivesWithTrailingTriviaAsync()
                 Assert.Equal(1, i);
             }
         }
+
+        [Fact]
+        public static async Task ReadCollectionPassingNullValueAsync()
+        {
+            using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes("null")))
+            {
+                IList<object> referenceTypeCollection  = await JsonSerializer.ReadAsync<IList<object>>(stream);
+                Assert.Null(referenceTypeCollection);
+            }
+
+            using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes("null")))
```

Comment:
Your choice, but usually when a test has two near-identical tests inside it, it's better off being two tests. That way if it fails it's obvious which one failed. Also, if for some reason one of them starts to fail and needs to be disabled, we don't need to disable both.

Reply:
An alternative approach could be to use `Theory` here and test the method that accepts the returning type as an argument. I would be fine with either approaches. If the intent is to actually test the generic overload (and substituting it with the non-generic API isn't feasible), then two separate tests works just as well.

```C#
[Theory]
[InlineData(typeof(IList<object>))]
[InlineData(typeof(IList<int>))]
public static async Task ReadSimpleObjectAsync(Type type)
{
    using (MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes("null")))
    {
        object collection = await JsonSerializer.ReadAsync(stream, type);
        Assert.Null(collection);
    }
}
```

Reply:
I will stick to create separate unit test methods for each scenario.

The reason is, the origin of this issue was using the generic type overload and also there are no significant differences between both signatures; as you can see, the two public methods end up calling the private overload:
https://github.com/dotnet/corefx/blob/eaa0079941923b4f903344130315f4cd332b84fe/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Stream.cs#L75